Added Lust's Ruin, and a magical whip that makes people orgasm.

SkullyRoberts opened the pull request:
[Changelogs]: # 
Add: Added the Lust Ruin, which is basically a sex dungeon in lavaland.
Add: added Lust's Whip, a whip that makes anyone who is hit by it (who has arousal enabled) orgasm.

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)


Because we have a bunch of the 7 deadly sin ruins, but noone ever made one for Lust. That changes now.
Besides, this is Citadel. We need to make as many sex dungeons as possible.

SkullyRoberts committed:
Added Lust Ruin

Made lust ruin, full of sex toys, bodily fluids, and more smut than you can imagine.
Added Lust's Whip, a whip that makes anyone you hit with it instantly orgasm.

SkullyRoberts committed:
added lust ruin area

SkullyRoberts committed:
Added the Lust Ruin Map template

Trilbyspaceclone commented on _maps/RandomRuins/LavaRuins/lavaland_surface_lust.dmm:
```diff
@@ -0,0 +1,60 @@
+"L" = (/obj/machinery/door/airlock/gold/glass,/obj/effect/mine/sound/bwoink,/turf/open/floor/mineral/plasma,/area/ruin/powered/lust)
+"M" = (/obj/effect/decal/cleanable/femcum,/obj/structure/closet,/obj/item/dildo/random,/obj/item/dildo/random,/obj/item/electropack/shockcollar,/turf/open/floor/mineral/plasma,/area/ruin/powered/lust)
+"N" = (/obj/structure/bodycontainer/morgue,/turf/open/floor/mineral/plasma,/area/ruin/powered/lust)
+"O" = (/turf/open/floor/plating/asteroid/basalt/lava_land_surface,/area/ruin/powered/lust)
```
Maybe not power tiles outside the ruin

Trilbyspaceclone commented on _maps/RandomRuins/LavaRuins/lavaland_surface_lust.dmm:
```diff
@@ -0,0 +1,60 @@
+"a" = (/turf/template_noop,/area/template_noop)
```
None of these have air do they?

Trilbyspaceclone replied:
If they do have are, what air truffs is it pulling?

SkullyRoberts replied:
They should take the turf and area of whatever tile it WOULD have replaced

Trilbyspaceclone commented on _maps/RandomRuins/LavaRuins/lavaland_surface_lust.dmm:
```diff
@@ -0,0 +1,60 @@
+"l" = (/obj/effect/decal/cleanable/femcum,/obj/item/toy/plush/random,/turf/open/floor/mineral/diamond,/area/ruin/powered/lust)
+"m" = (/obj/effect/decal/cleanable/femcum,/obj/effect/decal/cleanable/semen,/turf/open/floor/mineral/diamond,/area/ruin/powered/lust)
+"n" = (/obj/item/melee/chainofcommand/lust,/turf/open/floor/mineral/plasma,/area/ruin/powered/lust)
+"o" = (/obj/item/dildo/random,/turf/open/floor/mineral/plasma,/area/ruin/powered/lust)
```
Plasma floor tiles are a bad idea...

Trilbyspaceclone commented on code/datums/ruins/lavaland.dm:
```diff
@@ -222,4 +222,12 @@
+	description = "An abandoned sex dungeon with a powerful artifact."
+	suffix = "lavaland_surface_lust.dmm"
+	allow_duplicates = FALSE
+	cost = 10
```
Do to only being a lewd place with no real loot of worth It can be lowerd to 5

Trilbyspaceclone commented on code/datums/ruins/lavaland.dm:
```diff
@@ -222,4 +222,12 @@
+	cost = 5
+
+/datum/map_template/ruin/lavaland/lust
+	name = "Lust Ruin"
```
Ruin of Lust

Trilbyspaceclone commented on code/game/objects/items/melee/misc.dm:
```diff
@@ -29,6 +29,16 @@
 
+/obj/item/melee/chainofcommand/lust
+	name = "Lust's Whip"
+	desc = "A strange whip that sparks arousal from the slightest touch"
```
.

Trilbyspaceclone commented on code/game/objects/items/melee/misc.dm:
```diff
@@ -29,6 +29,16 @@
+	icon_state = "lustwhip"
+	damtype = AROUSAL
+	force = 100
+	attack_verb = list("punished", "disciplined", "tamed", "teased")
```
flailed, whipped 
Same as normal whip

Trilbyspaceclone reviewed (changes requested):
A few problems or nonblems I see

deathride58 reviewed (changes requested):
Use the map merger
also add inhands for the whip plz

SkullyRoberts committed:
mapmerge

SkullyRoberts committed:
Taking suggestions

Decreased cost to 5, added in-hand files, added "flailed" and "whipped" action verbs, in addition to the kinky ones.

Trilbyspaceclone commented on _maps/RandomRuins/LavaRuins/lavaland_surface_lust.dmm:
```diff
@@ -0,0 +1,471 @@
+/turf/open/floor/mineral/diamond,
+/area/ruin/powered/lust)
+"r" = (
+/obj/structure/bed/alien,
```
Alien beds are rather OP its why we dont have the crash on.

deathride58 commented on code/game/objects/items/melee/misc.dm:
```diff
@@ -29,6 +29,18 @@
+	desc = "A strange whip that sparks arousal from the slightest touch"
+	icon_state = "lustwhip"
+	lefthand_file = 'icons/mob/inhands/weapons/melee_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/melee_righthand.dmi'
```
Lacking inhand sprites in this file

SkullyRoberts replied:
Did it.

SkullyRoberts committed:
Set Lust's Whip's inhand icon to the chain

Just like every other whip
